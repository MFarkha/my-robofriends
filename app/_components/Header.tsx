import localFont from 'next/font/local';
 
// Font files can be colocated inside of `app`
const myFont = localFont({
    src: '../fonts/Robotronics.woff',
    display: 'swap',
    variable: '--font-robotronics',
    style: 'normal',
    weight: 'normal'
})
 
const Header = () => {
    return (
        <h1 className={`${myFont.className} f1`}>My Robofriends</h1>
    )
}

export default Header;