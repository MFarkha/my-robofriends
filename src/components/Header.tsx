import { Component } from "react";

class Header extends Component {
  shouldComponentUpdate() {
    return false;
  }
  render() {
    return (
      <>
        <h1 className="f1">My Robofriends</h1>
      </>
    );
  }
}

export default Header;
