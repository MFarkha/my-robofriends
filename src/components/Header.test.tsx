import Header from './Header';
import { shallow } from 'enzyme';

it('expects to render the Header component', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
});
it('is function: `shouldComponentUpdate` must always return false', () => {
    const wrapper = shallow(<Header/>);
    const headerInstance = wrapper.instance() as Header;
    expect(headerInstance.shouldComponentUpdate()).toBeFalsy();
});