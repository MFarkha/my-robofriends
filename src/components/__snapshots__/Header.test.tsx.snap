// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render the Header component 1`] = `
<Fragment>
  <h1
    className="f1"
  >
    My Robofriends
  </h1>
</Fragment>
`;
